book/src/rfcs/ffadd.md

```diff
@@ -0,0 +1,192 @@
+# RFC: Foreign Field Addition
+
+This document is meant to explain the foreign field addition gate in Kimchi.
+
+## Overview
+
+The goal of this gate is to perform the following operation between a left $a$ and a right $b$ input, to obtain a result $r$
+$$a + b = r \mod f$$
+where $a,b,r\in\mathbb{F}_f$ belong to the foreign field of modulus $f$ and we work over a native field $\mathbb{F}_n$ with modulus $n$. 
+
+If $f<n$ then we can easily perform the above computation. But in this gate we are interested in the contrary case in which $f>n$. In order to deal with this, we will divide foreign field elements into limbs that fit in our native field. We want to be compatible with the foreign field multiplication gate, and thus the parameters we will be using are the following:
+
+- 3 limbs of 88 bits each (for a total of 264 bits)
+- $2^{264} > f$
+- our foreign field will have 256-bit length
+- our native field has 255-bit length
+
+### Splitting the addition
+
+Let's take a closer look at what we have, if we split our variables in limbs (using little endian)
+
+```text
+bits  0..............87|88...........175|176...........263
+
+a  =  (-------a0-------|-------a1-------|-------a2-------)
++ 
+b  =  (-------b0-------|-------b1-------|-------b2-------)
+=
+r  =  (-------r0-------|-------r1-------|-------r2-------)  mod(f)
+```
+We will perform the addition in 3 steps, one per limb. Now, when we split long additions in this way, we must be careful with carry bits between limbs. Also, even if $a$ and $b$ are foreign field elements, it could be the case that $a+b$ is already larger than the modulus (such addition could be at most $2f-2$), and thus we will have to consider the more general case. That is,
+$$ a + b = q \cdot f + r \mod 2^{264}$$
+with a field overflow term $x$ that will be either $0$ or $1$. Looking at this in limb form, we have:
+
+```text
+bits  0..............87|88...........175|176...........263
+
+a  =  (-------a0-------|-------a1-------|-------a2-------)
++ 
+b  =  (-------b0-------|-------b1-------|-------b2-------)
+=
+o  =  0 | 1
+·
+f  =  (-------f0-------|-------f1-------|-------f2-------)
++
+r  =  (-------r0-------|-------r1-------|-------r2-------)
+```
+
+First, if $a+b$ was larger than $f$, then we will have a field overflow (represented by $o = 1$) and we will have to subtract $f$ from the sum $s = a+b$ to obtain $r$. The fact that we introduce this subtraction operation means that the limbs could have a carry bit, but also a borrow. Meaning, these flags can be anything in $\{-1, 0, 1\}$. Next we see more clearly how this works. 
```

Comment:
"could have a carry, but also a borrow"

"the carry flag can be anything in $\{-1, 0, 1\}$, where carry=-1 means a borrow."